test/src/test/java/org/corfudb/integration/CacheTest.java

```diff
@@ -0,0 +1,89 @@
+import org.corfudb.common.compression.Codec;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.corfudb.runtime.collections.CorfuTable;
+import org.corfudb.util.serializer.Serializers;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CacheTest extends AbstractIT {
+    private static final int divisor = 50;
+    private static final int iterations = 100;
+    private static final int numTables = 100;
+
+    @Test
+    public void cacheTest() throws IOException, InterruptedException {
+
+        Process server1 = new AbstractIT.CorfuServerRunner()
+                .setHost(DEFAULT_HOST)
+                .setPort(DEFAULT_PORT)
+                .setSingle(true)
+                .setLogPath("/tmp")
+                .runServer();
+        final int maxWriteSize = 500;
+        final String[] tableNames = new String[numTables];
+        Map<Integer, byte[]>[] maps = new Map[numTables];
+        Map<Integer, byte[]>[] maps_new = new Map[numTables];
+
+        CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                .builder()
+                .maxWriteSize(maxWriteSize)
+                .codecType(Codec.Type.NONE)
+                .build();
+
+        runtime = CorfuRuntime.fromParameters(params);
+        runtime.parseConfigurationString(DEFAULT_ENDPOINT);
+        runtime.connect();
+
+        for(int i=0; i<iterations; i++) {
+            maps[i] = runtime.getObjectsView()
+                    .build()
+                    .setTypeToken(new TypeToken<CorfuTable<Integer, byte[]>>() {})
+                    .setStreamName("table" + i)
+                    .serializer(Serializers.JAVA)
+                    .open();
+        }
+
+        byte[] smallPayload = new byte[maxWriteSize/divisor];
+
+        long start = System.currentTimeMillis();
+        for(int i=0; i<iterations; i++) {
+            runtime.getObjectsView().TXBegin();
+            maps[i].put(i, smallPayload);
+            runtime.getObjectsView().TXEnd();
+        }
+        long end = System.currentTimeMillis();
+        System.out.println("Total Write Time - " + (end - start));
+
+        // Verify that all those changes can be observed from a new client
+        CorfuRuntime rt1 = CorfuRuntime.fromParameters(params);
+        rt1.parseConfigurationString(DEFAULT_ENDPOINT);
+        rt1.connect();
+        for(int i=0; i<iterations; i++) {
+            maps_new[i] = runtime.getObjectsView()
+                    .build()
+                    .setTypeToken(new TypeToken<CorfuTable<Integer, byte[]>>() {})
+                    .setStreamName("table" + i)
+                    .serializer(Serializers.JAVA)
+                    .open();
+        }
+
+        start = System.currentTimeMillis();
+        for(int i=0; i<iterations; i++) {
+            assertThat(maps_new[i].get(i).equals(smallPayload));
+        }
```

Comment:
I would suggest to make the iterations bigger for one map, like 1 million. Also do the same for the put operation.

Reply:
Changed the test to iterate 10K times, each time writing to 100 tables in 1 transaction.  So the total writes will be    1 million